src/com/MessageRouter.java

```diff
@@ -39,16 +39,13 @@ public void run() {
         while (true) {
             if (!handleMessages.isEmpty()) {
                 MessageWithSocketChannel pollValue = (MessageWithSocketChannel) handleMessages.poll();
-                if (pollValue.getMessage().getType() == Message.Type.REGISTER) {
```

Comment:
You're right, but I can only comment where you change your code. Not in the whole project
I think I can try to use "Issues" for this goals...